packages/client/src/app/services/auth/auth.service.ts

```diff
@@ -250,6 +253,26 @@ export class AuthService {
       }));
   }
 
+  /**
+   * Sets the contentFilter cookie based on the values of the two provided booleans.
+   * 
+   * @param enableMature Enable mature check
+   * @param enableExplicit Enable explicit check
+   */
+  public setContentFilter(enableMature: boolean, enableExplicit: boolean) {
+    if (enableMature === true && enableExplicit === false) {
+      this.cookies.put('contentFilter', ContentFilter.MatureEnabled);
+    } else if (enableMature === false && enableExplicit === true) {
+      this.cookies.put('contentFilter', ContentFilter.ExplicitEnabled);
+    } else if (enableMature === true && enableExplicit === true) {
+      this.cookies.put('contentFilter', ContentFilter.Everything);
+    } else if (enableMature === false && enableExplicit === false) {
+      this.cookies.put('contentFilter', ContentFilter.Default);
+    }
+
+    location.reload();
```

Comment:
huh, do we really need to force a refresh before cookies get recognized?

Reply:
yeah, without a refresh the cookie doesn't get stored for some reason.